enhancements/worker-latency-profile/worker-latency-profile.md

```diff
@@ -0,0 +1,450 @@
+
+`Kube Controller Manager` will wait for 2 minutes to consider unhealthy status for the node and since `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds` of the Kube API Server are set to 60 seconds the total time will be 3 minutes before eviction process starts.
+
+
+| Component               | Flag Name                                | Flag Value  |
+| -----------             | -----------                              | ----------- |
+| Kubelet                 | --node-status-update-frequency           |    20s      |
+| Kube Controller Manager | --node-monitor-grace-period              |    2m       |
+| Kube API Server         | --default-not-ready-toleration-seconds   |    60s      |
+| Kube API Server         | --default-unreachable-toleration-seconds |    60s      |
+
+#### Low Update and Slow reaction
+
+Worker nodes may find themselves in a network with extremely high latency and/or bad reliability.
+In this scenario, we will reduce the frequency of `Kubelet` updates further to every 1 minute (`--node-status-update-frequency`) and change the node monitor grade period of the `Kube Controller Manager` to every 5 minutes (`--node-monitor-grace-period`). `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds` of the Kube API Server are set to 60 seconds.
+
+`Kube Controller Manager` will wait for 5 minutes to consider unhealthy status for the node and since `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds` of the `Kube API Server` are set to 60 seconds the total time will be 6 minutes before eviction process starts.
+
+
+| Component               | Flag Name                                | Flag Value  |
+| -----------             | -----------                              | ----------- |
+| Kubelet                 | --node-status-update-frequency           |    1m       |
+| Kube Controller Manager | --node-monitor-grace-period              |    5m       |
+| Kube API Server         | --default-not-ready-toleration-seconds   |    60s      |
+| Kube API Server         | --default-unreachable-toleration-seconds |    60s      |
+
+
+#### Failure Modes
+1. In case of failure `WorkerLatencyProfileStatus` should point towards failed component(s)
+
+
+### Cluster Stability Analysis
+
+Let's try to understand the effect of applying various worker latency profiles on the overall stability of the cluster.
+
+Kubelet will try to make [5 post attempts](https://github.com/kubernetes/kubernetes/blob/release-1.23/pkg/kubelet/kubelet.go#L129) to update it's status. `--node-status-update-frequency` determines the frequency of attemps to update status.
```

Comment:
In the default installation, kubelet will...

Reply:
When it comes to number of  attempts, the value is not configurable in the kubelet. It's [hard coded](https://github.com/kubernetes/kubernetes/blob/release-1.23/pkg/kubelet/kubelet.go#L129) to 5 attempts in the source code. So I am hesitant to state `In the default installation`, because it's unconfigurable for default or non-default configuration of the Kubelet.